server/resp/src/main/java/org/infinispan/server/resp/commands/hll/PFADD.java

```diff
@@ -0,0 +1,87 @@
+package org.infinispan.server.resp.commands.hll;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.server.resp.hll.HyperLogLog;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * The `<code>PFADD key [element [element ...]]</code>` command.
+ * <p>
+ * Adds an element to the HyperLogLog (HLL) structure. The command always verify if is possible to use the HLL structure
+ * with the given <code>key</code>. If the command is invoked without elements, the entry is created.
+ * </p>
+ *
+ * @see <a href="https://redis.io/commands/pfadd/">Redis documentation.</a>
+ * @since 15.0
+ * @author José Bolina
+ */
+public class PFADD extends RespCommand implements Resp3Command {
+
+   public PFADD() {
+      super(-2, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx, List<byte[]> arguments) {
+      byte[] key = arguments.get(0);
+      AdvancedCache<byte[], Object> cache = handler.cache().withMediaType(MediaType.APPLICATION_OCTET_STREAM, null);
+      AtomicReference<UpdateStatus> status = new AtomicReference<>();
+      CompletionStage<UpdateStatus> cs = cache.computeAsync(key, (ignore, stored) -> {
+         HyperLogLog hll = parseHLL(stored);
+
+         // Storing something which is not an HLL.
+         if (hll == null) {
+            RespErrorUtil.wrongType(handler.allocator());
+            status.set(UpdateStatus.NO_OP);
+            return stored;
+         }
+
+         List<byte[]> elements = arguments.subList(1, arguments.size());
+         long before = hll.cardinality();
+         for (byte[] el : elements) {
+            hll.add(el);
+         }
+         status.set(stored == null || before < hll.cardinality() ? UpdateStatus.ADDED : UpdateStatus.NO_CHANGE);
+         return hll;
+      }).thenApply(ignore -> status.get());
+
+      return handler.stageToReturn(cs, ctx, (res, buf) -> {
+         switch (res) {
```

Comment:
What about NO_OP? Looks like we aren't sending a response in that case?

Reply:
Past self thought it would make sense to reply from within the lambda :sweat_smile:. Handling it now.